arguments.py

```diff
@@ -99,10 +103,44 @@ def initArgs():
 	parser.add_argument("-p", "--processes", type=int, default=-1,
 						help="Number of processes to use for rendering and export. Default: Number of processor cores (or hyperthreads, if supported)")
 
+
+	# Parse arguments once to get preset flag
+	args = parser.parse_args()
+
+	# Count number of positional arguments explicitly set
+	posArgs = 0
+	for arg in sys.argv:
+		if arg[0] == '-':
+			break
+		posArgs += 1
```

Comment:
This is a super ugly solution but, at least with the current approach to parse presets, I can't think of a better way to do this.
The problem is that the output directory must be the 2nd argument **if it is set** but it might not be there as it is an optional. The current code just assumes that the first argument that starts with a '-' must be a flag, but that is not true as the output folder might start with a '-' as well.

Reply:
Not necessary since positional arguments only need to preserve relative order to each other, no particular ordering w.r.t. flags necessary. Thanks to @irgendwr